File: slice/119_get_row.go
package slice

func getRow(rowIndex int) []int {
	dp := make([]int, rowIndex+1)
	for i := range dp {
		dp[i] = 1
	}
	for i := 1; i <= rowIndex; i++ {
		for j := i - 1; j > 0; j-- {
			dp[j] += dp[j-1]
		}
	}
	return dp
}

File: slice/119_get_row.rs
impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let mut dp = vec![1; (row_index + 1) as usize];
        for i in 1..=row_index as usize {
            for j in (1..i).rev() {
                dp[j] += dp[j - 1];
            }
        }
        dp
    }
}
